modules/imgproc/src/distransform.cpp

```diff
@@ -836,12 +836,13 @@ void cv::distanceTransform( InputArray _src, OutputArray _dst, OutputArray _labe
             }
 #endif
 
+            temp.create(size.height + border*2, size.width + border*2, CV_32SC1);
             distanceTransform_3x3(src, temp, dst, _mask);
         }
         else
         {
 #if defined (HAVE_IPP) && (IPP_VERSION_X100 >= 700)
-            CV_IPP_CHECK()
+            if (dst.isContinuous() && CV_IPP_CHECK_COND)
```

Comment:
Need to check how it worked with ROI dst before.

Reply:
@opencv-alalek, @asmorkalov, I have verified that this change breaks IPP support for a valid usecase with `dst` as a ROI. `isContinuous` is false but this is not the same if `Mat` is not continuous with an irregular step for large buffer.

Am I right that in case of `.create` called on an empty `Mat` but with a large sizes (for example, 70000x70000), Mat memory is constructed from a multiple continuous blocks and step is irregular?

Reply:
probably related to this discussion: https://github.com/opencv/opencv/issues/12018. Changed check to INT_MAX overflow

Reply:
the right link about `isContinuous()` mess: #11505